packages/fields/src/Block.js

```diff
@@ -32,15 +38,22 @@ export class Block {
     return [];
   }
 
-  get path() {
-    throw new Error(`${this.constructor.name} must have a 'path' getter`);
+  getFieldDefinitions() {
+    return {};
   }
 
-  get fieldDefinitions() {
-    return {};
+  /**
+   * @param graphQlArgs {Object}
+   */
+  getMutationOperationResults() {
+    return;
   }
 
-  extendAdminMeta(meta) {
-    return meta;
+  getAdminViews() {
```

Comment:
Not sure if there's a better name for this but when I was reading through this, it wasn't totally clear to me how the views that `getAdminViews` return are used on the client. Eventually I looked at the implementation of the Content field type Controller and understood that they're all interpreted as blocks on the client but that seemed strange because it's a very different meaning to `extendViews`/now `extendAdminViews` on field Controllers where it's an object and each view on the object is used by its own thing but the names are really similar.

Reply:
Yeah, I got stuck on naming this one for that exact reason. There's another wrinkle: the paths are loaded and interpreted as Blocks _if they have an export of `type`_, otherwise they're just taking advantage of the `import()` mechanism.

Open to suggestions (with minimal bike shedding) 👍

Reply:
> if they have an export of type, otherwise they're just taking advantage of the import() mechanism.

What's an example of something that would be here and wouldn't be a block?

Reply:
This one: https://github.com/keystonejs/keystone-5/pull/1307/files#diff-c850d4389f69a80bcd0223eee6a2cf48

Used here: https://github.com/keystonejs/keystone-5/pull/1307/files#diff-eef8d04a752abe36a598e4cdca7cbdb4R66

And loaded here: https://github.com/keystonejs/keystone-5/pull/1307/files#diff-f72926d16a4c17fb27bd1aa14ec46339R24

Reply:
Hmmmmmmm, I don't really like that the path has to be specified in the options and getAdminViews but I can't think of a better way to do all this right now so I guess this is all fine then.